polaris-core/src/main/java/org/apache/polaris/core/storage/FileStorageConfigurationInfo.java

```diff
@@ -53,4 +54,15 @@ public void validatePrefixForStorageType(String loc) {
               "Location prefix not allowed: '%s', expected prefix: file:// or / or *", loc));
     }
   }
+
+  public @Nonnull FileStorageConfigurationInfo merge(@Nonnull FileStorageConfigurationInfo other) {
+    if (other.getStorageType() != this.getStorageType()) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Storage type mismatch: %s vs %s", this.getStorageType(), other.getStorageType()));
+    }
```

Comment:
Minor: we could use precondition here.
```
    Preconditions.checkArgument(other.getStorageType() == this.getStorageType(),
          String.format(
              "Storage type mismatch: %s vs %s", this.getStorageType(), other.getStorageType()));
```

Reply:
Thanks for the suggestion, will revise my PR with `Preconditions`.